src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -58,10 +58,21 @@ System::Clock::Timeout BDXDownloader::GetTimeout()
     return mTimeout;
 }
 
-void BDXDownloader::Reset()
+void BDXDownloader::Reset(bool resetBdxTransfer, bool abortImageProcessing, OTAChangeReasonEnum reason)
```

Comment:
This is too complicated and has too many conditions. Callers should release resources if needed, or there should be a way not to have to pass 3 arguments that are hard to audit every site.

Reply:
Agree, this is confusing, added a private API `CleanupOnError()`, PTAL!